tests/common/plugins/conditional_mark/__init__.py

```diff
@@ -126,30 +140,31 @@ def load_basic_facts(session):
 
     return results
 
-
-def find_longest_match(nodeid, case_names):
+def find_longest_match(nodeid, conditions):
     """Find the longest match of the given test case name in the case_names list.
 
     This is similar to longest prefix match in routing table. The longest match takes precedence.
 
     Args:
         nodeid (str): Full test case name
-        case_names (list): List of test case names
+        conditions (list): List of conditions
 
     Returns:
         str: Longest match test case name or None if not found
     """
-    longest_match = None
+    longest_match = []
     max_length = -1
-    for case_name in case_names:
-        if nodeid.startswith(case_name):
-            length = len(case_name)
+    for condition in conditions:
+        if nodeid.startswith(condition.keys()[0]):
+            length = len(condition)
             if length > max_length:
                 max_length = length
-                longest_match = case_name
+                longest_match = []
+                longest_match.append(condition)
+            elif length == max_length:
```

Comment:
I think this algorithm is wrong. During the iteration, the max_length may change. It is not determined yet.